reading_comprehension/qanet_for_drop.py

```diff
@@ -0,0 +1,330 @@
+        embedded_question = self._highway_layer(self._embedding_proj_layer(embedded_question))
+        embedded_passage = self._highway_layer(self._embedding_proj_layer(embedded_passage))
+
+        batch_size = embedded_question.size(0)
+
+        projected_embedded_question = self._encoding_proj_layer(embedded_question)
+        projected_embedded_passage = self._encoding_proj_layer(embedded_passage)
+
+        encoded_question = self._dropout(self._phrase_layer(projected_embedded_question, question_mask))
+        encoded_passage = self._dropout(self._phrase_layer(projected_embedded_passage, passage_mask))
+
+        # Shape: (batch_size, passage_length, question_length)
+        passage_question_similarity = self._matrix_attention(encoded_passage, encoded_question)
+        # Shape: (batch_size, passage_length, question_length)
+        passage_question_attention = masked_softmax(passage_question_similarity, question_mask)
+        # Shape: (batch_size, passage_length, encoding_dim)
+        passage_question_vectors = util.weighted_sum(encoded_question, passage_question_attention)
+
+        # Shape: (batch_size, question_length, passage_length)
+        question_passage_attention = masked_softmax(passage_question_similarity.transpose(1, 2), passage_mask)
+        # Shape: (batch_size, passage_length, passage_length)
+        attention_over_attention = torch.bmm(passage_question_attention, question_passage_attention)
+        # Shape: (batch_size, passage_length, encoding_dim)
+        passage_passage_vectors = util.weighted_sum(encoded_passage, attention_over_attention)
+
+        # Shape: (batch_size, passage_length, encoding_dim * 4)
+        merged_passage_attention_vectors = self._dropout(
+                torch.cat([encoded_passage, passage_question_vectors,
+                           encoded_passage * passage_question_vectors,
+                           encoded_passage * passage_passage_vectors],
+                          dim=-1)
+        )
+
+        modeled_passage_list = [self._modeling_proj_layer(merged_passage_attention_vectors)]
+
+        for _ in range(4):
+            modeled_passage = self._dropout(self._modeling_layer(modeled_passage_list[-1], passage_mask))
+            modeled_passage_list.append(modeled_passage)
+
+        passage_weights = self._passage_weights_predictor(modeled_passage_list[4]).squeeze(-1)
+        passage_weights = passage_weights * passage_mask
+        passage_vector = util.weighted_sum(modeled_passage_list[4], passage_weights)
+        question_weights = self._question_weights_predictor(encoded_question).squeeze(-1)
+        question_weights = question_weights * question_mask
+        question_vector = util.weighted_sum(encoded_question, question_weights)
+
+        # Shape: (batch_size, 3)
+        answer_type_logits = self._answer_type_predictor(torch.cat([passage_vector, question_vector], -1))
+        answer_type_probs = torch.nn.functional.softmax(answer_type_logits, -1)
```

Comment:
Is it possible to track accuracy on this?  That'd be helpful.

Reply:
Ok, I will write some codes to track this.  For analysis, I have printed some results for this, it seems that the model predict all types as `span answer`.